lib/matplotlib/tests/test_widgets.py

```diff
@@ -59,6 +59,46 @@ def test_rectangle_selector():
     check_rectangle(props=dict(fill=True))
 
 
+@pytest.mark.parametrize('spancoords', ['data', 'pixels'])
+@pytest.mark.parametrize('minspanx, x1', [[0, 10], [1, 10.5], [1, 11]])
+@pytest.mark.parametrize('minspany, y1', [[0, 10], [1, 10.5], [1, 11]])
+def test_rectangle_minspan(spancoords, minspanx, x1, minspany, y1):
+    ax = get_ax()
+    ax._n_onselect = 0
```

Comment:
AFAICS this dynamically adds an attribute. If so, this needs explanation:
```suggestion
    # add an attribute to track the number of onselect calls
    ax._n_onselect = 0
```
OTOH you are only working with a single axis. Couldn't you just use a local variable?

Reply:
I tried a local variable and it gives
```python
    def onselect(epress, erelease):
>       n_onselect += 1
E       UnboundLocalError: local variable 'n_onselect' referenced before assignment
```

I've added a comment though

Reply:
That would probably need a `nonlocal` statement. But I can live with a comment as well.

Reply:
Seems like a use case for `Mock`? Then `assert_not_called`, `assert_called_once_with`, etc.

Reply:
I'm happy to look into mocking, but would advocate for this PR to go in first so https://github.com/matplotlib/matplotlib/pull/21945 can be tested with these tests, and because the current code is how other widget tests do similar checks.